config.go

```diff
@@ -161,6 +174,10 @@ func (l *Loader) resolve(ctx context.Context, s *StructConfig) error {
 		if u, ok := b.(Unmarshaler); ok {
 			err := u.Unmarshal(ctx, s.S)
 			if err != nil {
+				if uerr, ok := err.(*file.ErrOpenOptionalFile); ok {
```

Comment:
I'm not entirely sure that importing another backend package from the top-level package is a good idea. Also, is it really worth logging the error in the common case that the file isn't present and it's optional?

How about, instead of creating a new error type, making the file backend slightly more careful about its error checking. If the file isn't found (check the error with os.IsNotExist), then return backend.ErrNotFound; otherwise return an error as usual?

Then there's no need to log the error, so no need for the logger, and an unexpected error (e.g. permission denied) will cause a failure, which is probably what you want.

Reply:
Yes I agree I will remove the logger and using the backend.ErrNotFound when the file is not found is totally coherent.

Reply:
Done in ddfd957